src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -61,16 +61,19 @@ int ObjectCacheStore::init(bool reset) {
   // TODO(dehao): fsck and reuse existing cache objects
   if (reset) {
     std::error_code ec;
-    if (efs::exists(m_cache_root_dir)) {
-      // remove all sub folders
-      for (auto& p : efs::directory_iterator(m_cache_root_dir)) {
-        efs::remove_all(p.path());
-      }
-    } else {
-      if (!efs::create_directories(m_cache_root_dir, ec)) {
-        lderr(m_cct) << "fail to create cache store dir: " << ec << dendl;
-        return ec.value();
+    try {
+      if (efs::exists(m_cache_root_dir)) {
+        // remove all sub folders
```

Comment:
Due to this I don't like very much that the default `immutable_object_cache_path` is `/tmp`. Consider an admin just learning about this tool and starting with defaults. I think she would not be happy to learn that all subfolders in '/tmp' dir had gone. This is actually what I experienced when first tried it. Luckily it was not from the root so eventually it failed due to a permission error, but who knows what folders it had removed before this... 

Something like `/tmp/ceph/immutable_object_cache` would be much safer.

Reply:
Good point. Switched to `/tmp/ceph_immutable_object_cache`